internal/configs/test_file.go

```diff
@@ -839,6 +846,7 @@ var testRunOptionsBlockSchema = &hcl.BodySchema{
 		{Name: "refresh"},
 		{Name: "replace"},
 		{Name: "target"},
+		{Name: "state_alias"},
```

Comment:
I think I wouldn't put this within the run options block. The run options are a 1:1 mapping to actual run options a user can specify on the command line to the plan command, and overriding the state is not something that can be done this way. I think putting it here confuses things.

I think we can actually just have it at the root level of the `run` block. It's metadata about the run block rather than something that changes the behaviour of the plan/apply operation itself.